modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/related/models/test/ObjectRelatedModelsProviderTest.java

```diff
@@ -957,6 +946,103 @@ private void _testSystemObjectEntry1toMObjectRelatedModels()
 		}
 	}
 
+	private void _testSystemObjectEntry1toMObjectUnrelatedModels(long companyId)
+		throws Exception {
+
+		User user = UserTestUtil.getAdminUser(companyId);
+
+		Assert.assertNotNull(user);
+
+		long originalCompanyId = CompanyThreadLocal.getCompanyId();
+		String originalName = PrincipalThreadLocal.getName();
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		CompanyThreadLocal.setCompanyId(companyId);
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(user));
+		PrincipalThreadLocal.setName(user.getUserId());
```

Comment:
@murilostodolni can we change here to `try/finally` to make sure we restore the original state no matter what happens on the test?

Reply:
@murilostodolni you can also use
```
try (SafeCloseable safeCloseable =
		CompanyThreadLocal.setWithSafeCloseable(companyId)) {
```
for `CompanyThreadLocal`